lib/logger.js

```diff
@@ -522,83 +614,104 @@ class Logger extends EventEmitter {
       ? buffer[buffer.length - 1].line
       : null
 
-    axios(config)
-      .then((response) => {
-        // We have a 200-level success code.
-        const totalLinesSent = buffer.length
-        this[kIsLoggingBackedOff] = false
-        this[kAttempts] = 0
-        this[kIsSending] = false
-        this[kTotalLinesReady] -= totalLinesSent
-        this[kReadyToSend].shift()
-
-        // Assist GC by killing the buffer and removing it from readyToSend
-        buffer.length = 0
-
-        this.emit('send', {
-          httpStatus: response.status
-        , firstLine
-        , lastLine
-        , totalLinesSent
-        , totalLinesReady: this[kTotalLinesReady]
-        , bufferCount: this[kReadyToSend].length
-        })
-
-        if (this[kReadyToSend].length) {
-          // Continue to send any backed up payloads that have accumulated
-          this.send()
-          return
-        }
-
-        this.emit('cleared', {
-          message: ALL_CLEAR_SENT
-        })
-      })
-      .catch((error) => {
-        const code = error.response
-          ? error.response.status
-          : error.code // timeouts will populate this
+    const data = stringify({e: 'ls', ls: buffer})
 
-        const retrying = ERROR_CODES_TO_RETRY.has(code)
-        const err = new Error('An error occured while sending logs to the cloud.')
+    this._getSendPayload(data, (error, payload) => {
```

Comment:
The code inside the callback is all the same, aside from the top part.  There was no cleaner way to do this.  I had a streams version, but there's a problem with axios/nock in there somewhere.  A future PR will need to get off of axios for something like `got`.